### PR TITLE
GAC model for the UI

### PR DESCRIPTION
**Pull request by hunterbarber**
## Fixes/Resolves:
Adds a basic single contactor (or multiple contactors in parallel) GAC system to the UI. Includes the flowsheet, UI, and image files. Also require UI testing generatlization.

Still TODO's for this or potentially other PR's
- [x] Add availability to build for specific solute contaminant species
- [x] Allow for the variable GAC config options
- [x] Test files
The following were added as TODOs in the flowsheet code, not prioritized for first implementation
- [ ] Check robustness of solves
- [ ] (Ambitious) Allow for lead/lag build option, and corresponding flowsheet image
- [ ] (Later) Add surrogates for less parameterization

## Summary/Motivation:
Added flowsheets to the UI (#1117)

## Changes proposed in this PR:
- Add basic GAC flowsheet generalized with a "solute" contaminant
- Add basic GAC UI wrapper for the flowsheet
- Add GAC UI image
- Generalize UI testing for flowsheets with build options

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
4. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by ksbeattie** [comment]
Related to #1239

**Comment by hunterbarber** [comment]
To pass tests on the [GAC UI](https://github.com/watertap-org/watertap/pull/1238), I have to add an if statement into `gac_ui.build_flowsheet` because `test_build` in `test_flowsheet_interfaces.py` doesn't pass a `build_options` when testing (`build_options` remains `None`). However without the if statement (how it currently is in the PR linked), the actual GUI works fine because the UI always does pass `build_options `and the user is forced to build prior to solving. Is this expected? I saw the only other UI using build options is the `nf_ui.py` and handles it a little different, but it is not very clean and looks like it maybe was just done to manage tests.

EDIT: It is not expected. I'm adding some UI testing generalization in this PR to manage this in future UI fs with `build_options`.

**Comment by hunterbarber** [comment]
Cancelled workflow for December pending PRs to get priority, just need to merge main and run workflow once others in queue are finished.

**Comment by hunterbarber** [comment]
Since its assumed for the UI that a user with know what solute they're designing for, I just picked moderate values for solute properties to initialize the model. I figured those selected initial values were good but It's failing macOS. I don't really know how I can confirm and debug it (or whether I should for a macOS fail). @adam-a-a @bknueven are maybe good to ask for any recommendations? I don't know what we are typically doing now when only a macOS fail comes up, if I remember correctly we no longer just mark that it requires watertap-ipopt(?) but let me know.

**Comment by ksbeattie** [comment]
This might require a similar patch as was done in #1185

**Comment by hunterbarber** [comment]
> This might require a similar patch as was done in #1185

For visibility, this isn't equivalent. These tests were failing `unit_ui.py` and not `unit.py`

**Comment by MarcusHolly** [comment]
Hmmm, yea you're right... ig I haven't run into this issue b4 then, so I'm not sure what the "solution" is.

File: watertap/examples/flowsheets/gac/gac_ui.py [hunk 1]
```diff
@@ -0,0 +1,816 @@
...
+
+__author__ = "Hunter Barber"
+
+
+def export_to_ui():
+    """
+    Exports the variables, flowsheet build, and solver results to the GUI.
+    """
+    return FlowsheetInterface(
+        name="GAC",
```
Comment (MarcusHolly) [inline watertap/examples/flowsheets/gac/gac_ui.py]: I wonder if it's worth spelling GAC out here...

**Review by MarcusHolly — APPROVED** [review]
LGTM - just one minor comment, but I don't think it necessarily needs to be changed.

**Comment by hunterbarber** [comment]
Probably could still use one reviewer for the technical content (@adam-a-a, @kurbansitterley) and one reviewer for the changes to the UI fsapi and testing (@MichaelPesce, @dangunter)

**Review by kurbansitterley — APPROVED** [review]
Technical GAC-related content looks pretty straightforward. LGTM.

**Comment by ksbeattie** [comment]
@MichaelPesce will review this with a closer look at the UI running other models.

**Review by MichaelPesce — APPROVED** [review]
tested building and solving the other models on the UI using the changes in fsapi.py, everything looked good ✔️

**Pull request merged**